test-it/ViaTests/CheckValueNeedsUpdateTest.via

```diff
@@ -0,0 +1,60 @@
+
+define (MyVI1 dv(.VirtualInstrument (
+    Locals: c(
+        de(.Double myDouble)
```

Comment:
Add a test for an item that is not marked "de" and written to and make sure CheckValueNeedsUpdate returns false.

Reply:
Add a test for a cluster for the following cases:
1. cluster marked "de" and individual elements marked "e" and
        - write to the whole cluster and CheckValueNeedsUpdate
        - write to an element in the cluster and CheckValueNeedsUpdate
2. cluster marked "e" and some elements marked "de" and some marked "e"
        - write to the whole cluster and CheckValueNeedsUpdate
        - write to an "e" element in the cluster and CheckValueNeedsUpdate
        - write to an "de" element in the cluster and CheckValueNeedsUpdate
3. Write to an element of a data item in a Vireo data type such as ErrorCluster (or preferably AnalogWaveform) in VIA but internally defined as a cluster in Vireo and check for CheckValueNeedsUpdate

If some of the cases are not possible and need not be handled and tested because DFIR emission will not emit those cases, please document this somewhere (or close to where "de" is defined in the code) and in DFIR side PR.

Reply:
Added test for value not marked `de` and not marked `de` for double, array, cluster, Error, and AnalogWaveform<Double>.

I did not write tests for sub-elements of a cluster in the dataspace being marked as dataitems as dfir gen will not emit the dataitem marker for sub-elements of a cluster. Added a comment to the `de` definition stating this. I'll add a similar comment to the ASW side.